src/clients/KubectlClient.ts

```diff
@@ -101,10 +100,71 @@ export class KubectlClient implements IClient {
+            const kubeconfigPath: string = await this._accountContextManager.getKubeconfigPathAsync();
+            const args: string[] = [`get`, `pod`, podName, `-o`, `jsonpath="{.spec['containers','initContainers'][*].name}"`];
+            const kubectlOutput: string = await this.runKubectlCommandAsync(args, kubeconfigPath);
+            let containersList: string[];
+            if (kubectlOutput) {
+                containersList = kubectlOutput.replace('"', '').replace('"', '').split(' ');
+                return containersList.filter(s => !knownSideCars.find(knownSideCar => knownSideCar == s));
+            } else {
+                this._logger.error(TelemetryEvent.KubectlClient_GetContainerListError, new Error(`container list is null`));
+                return null;
+            }
+        } catch (error) {
+            this._logger.error(TelemetryEvent.KubectlClient_GetContainerListError, error);
+            return null;
+        }
+
+    }
+
+    public async getPodName(serviceName: string): Promise<string> {
+        try {
+            const kubeconfigPath: string = await this._accountContextManager.getKubeconfigPathAsync();
+            // find the pod ipaddress for the selected service
+            const args: string[] = [`get`, `ep`, serviceName, `-o`, `jsonpath='{.subsets[*].addresses[*].ip}'`];
+            let output = await this.runKubectlCommandAsync(args, kubeconfigPath);
+            // if there replicas and multiple pods, split it by apostrope, space example value: ''10.2.45.6 10.5.6.809''
+            // handles single value as well example value: ''10.56.78.90''
+            let ipaddress: string[];
+            if (output && output.length > 0) {
+                ipaddress = output.replace(/'/g, '').replace(/ /g, ',').split(',');
+            }
+
+            if (ipaddress.length > 0) {
+                // find the podname by looping the ip address
+                for (const ip in ipaddress) {
+                    const args2: string[] = [`get`, `pods`];
+                    const fieldSelector = '--field-selector=status.podIP='.concat(ipaddress[ip]);
+                    args2.push(fieldSelector);
+                    args2.push("-o=name");
+                    const finalOutput = await this.runKubectlCommandAsync(args2, kubeconfigPath);
+                    // split the output by forward slash and new line char at the end example value: 'pod/stats-api-ff7d66c5b-4nc9x\n' 
+                    // output is stats-api-ff7d66c5b-4nc9x
+                    if (finalOutput && finalOutput.length > 0) {
+                        return finalOutput.indexOf("/") != -1 ? finalOutput.split("/")[1].split("\n")[0] : finalOutput.split("\n")[0];
+                    }
+                }
+            }
+            return null;
+        } catch (error) {
+            this._logger.error(TelemetryEvent.KubectlClient_GetPodNameError, error);
```

Comment:
For some reason if it fails to get the containers we are logging and swallowing the exception to give the users chance to proceed further (in case of workloads which are not multi container can still use bridge). Hopefully I am making sense. in case of multiple containers, it would fail at later step where bridge CLI would not identify the container and fail.

Reply:
@Tatsinnit all good with this? These will be changing soon in refactoring PR which would introduce Error able interface similar to aks extension.

Reply:
Hiya!! Indeed!! Thanks @hsubramanianaks - errorable is something we have at many places, I think you and Pete might already be at sync on this) fyi - @peterbom , we should think of next level pkg for this but it can come as side project.